pyls/rpc_manager.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2017 Palantir Technologies, Inc.
+import logging
+from uuid import uuid4
+
+from concurrent.futures import ThreadPoolExecutor, Future
+from jsonrpc.base import JSONRPCBaseResponse
+from jsonrpc.jsonrpc1 import JSONRPC10Response
+from jsonrpc.jsonrpc2 import JSONRPC20Response, JSONRPC20Request
+from jsonrpc.exceptions import JSONRPCMethodNotFound, JSONRPCDispatchException, JSONRPCServerError
+
+log = logging.getLogger(__name__)
+
+RESPONSE_CLASS_MAP = {
+    "1.0": JSONRPC10Response,
+    "2.0": JSONRPC20Response
+}
+
+
+class MissingMethodException(Exception):
+    pass
+
+
+class JSONRPCManager(object):
+
+    def __init__(self, message_manager, message_handler):
+        self._message_manager = message_manager
+        self._message_handler = message_handler
+        self._shutdown = False
+        self._sent_requests = {}
+        self._received_requests = {}
+        self._executor_service = ThreadPoolExecutor(max_workers=5)
```

Comment:
Perhaps a comment that 5 is completely arbitrary and is reasonable to reconsider in the future. Else someone might dig around for the reason for 5